opacus/tests/scheduler_test.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import unittest
+
+import torch
+from torch import nn, optim
+from torch.utils.data import DataLoader, TensorDataset
+from opacus import PrivacyEngine
+from opacus.scheduler import ExponentialNoise, StepNoise, LambdaNoise
+
+
+class SchedulerTest(unittest.TestCase):
+    def setUp(self):
+        n_data, dim = 100, 10
+        data = torch.randn(n_data, dim)
+        model = nn.Linear(10, 10)
+        optimizer = optim.SGD(model.parameters(), lr=0.1)
+        data_loader = DataLoader(TensorDataset(data), batch_size=10)
+        self.engine = PrivacyEngine()
+
+        self.module, self.optimizer, self.data_loader = self.engine.make_private(
+            model, optimizer, data_loader, noise_multiplier=1.0, max_grad_norm=1.0
+        )
+
+    def test_exponential_scheduler(self):
+        gamma = 0.99
+        scheduler = ExponentialNoise(self.optimizer, gamma=gamma)
+
+        self.assertTrue(self.optimizer.noise_multiplier == 1.0)
+        scheduler.step()
+        self.assertTrue(self.optimizer.noise_multiplier == gamma)
+
+    def test_step_scheduler(self):
+        gamma = 0.1
+        step_size = 2
+        scheduler = StepNoise(self.optimizer, step_size=step_size, gamma=gamma)
+
+        self.assertTrue(self.optimizer.noise_multiplier == 1.0)
+        scheduler.step()
+        scheduler.step()
+        self.assertTrue(self.optimizer.noise_multiplier == gamma)
+        scheduler.step()
+        scheduler.step()
+        self.assertTrue(self.optimizer.noise_multiplier == gamma ** 2)
```

Comment:
I think we should add extra asserts in between steps: it's pretty fascinating coincidence, that this exact setup (step_size=2, only check after two steps) didn't catch the bug from the comment above